npm/ng-packs/packages/oauth/src/lib/strategies/auth-code-flow-strategy.ts

```diff
@@ -12,6 +12,18 @@ export class AuthCodeFlowStrategy extends AuthFlowStrategy {
 
     return super
       .init()
+      .then(() => {
+        this.oAuthService.events
+          .pipe(
+            filter(event => event.type === 'token_received'),
+            tap(e => {
+              const urlParams = window.location.search;
+              const uiCulture = new URLSearchParams(urlParams).get('ui-culture');
+              this.sessionState.setLanguage(uiCulture);
+            }),
+          )
+          .subscribe();
```

Comment:
Looks like we are using promise in observable this is not good way, I think we can track event out of promise chain. I belive we can move this logic to different method. Also we can call this method in init method.